src/SimSpace/SimFormat.hs

```diff
@@ -370,13 +383,85 @@ reformat programLines =
-               -> [Block]
-    reassemble nonImports (chunkedImports, leftovers)
-      = fmap Left nonImports
-     <> (fmap.fmap) renderImportStmts chunkedImports
-     <> leftovers
+    reassemble
+      :: (MonadIO m)
+      => [String]
+      -> ([Either BlankLine SortedImportStmts], [Block])
+      -> m [Block]
+    reassemble nonImports (chunkedImports, leftovers) = do
+        rechucked <- (if regroup then rechunk else pure) chunkedImports
+        pure $
+          fmap Left nonImports
+          <> (fmap . fmap) renderImportStmts rechucked
+          <> leftovers
+      where
+        rechunk
+          :: (MonadIO m)
+          => [Either BlankLine SortedImportStmts]
+          -> m [Either BlankLine SortedImportStmts]
+        rechunk ci = do
+          x@(preludes, locals, others) <-
+            foldlM
+              insertCatagorized
+              ([], [], [])
+              (foldMap fromSortedImportStmts [ stmts | Right stmts <- ci ])
+          pure $
+            [
+              Right $ toSortedImportStmts preludes,
+              Left "",
+              Right $ toSortedImportStmts others,
+              Left "",
+              Right $ toSortedImportStmts locals,
+              Left ""
+            ]
+
+        insertCatagorized
+          :: (MonadIO m)
+          => ([ImportStmt], [ImportStmt], [ImportStmt])
+          -> ImportStmt
+          -> m ([ImportStmt], [ImportStmt], [ImportStmt])
+        insertCatagorized (preludes, locals, others) stmt =
+          categorize stmt >>= \case
+            Prelude -> pure (preludes <> [stmt], locals, others)
+            Local -> pure (preludes, locals <> [stmt], others)
+            Other -> pure (preludes, locals, others <> [stmt])
+        
+        categorize :: (MonadIO m) => ImportStmt -> m ImportCategory
+        categorize stmt
+          | importStmtModuleName stmt `contains` "Prelude" =
+              pure Prelude
+          | otherwise =
+              liftIO (
+                readProcessWithExitCode
+                  "bash"
+                  [
+                    "-c",
+                    "stack exec ghc-pkg -- find-module "
+                    <> importStmtModuleName stmt
+                    <> " | grep -v 'pkgdb$' | grep -v 'package.conf.d$' | grep -qv '(no packages)'"
+                  ]
+                  ""
+              ) >>= \case
+                (ExitSuccess, _, _) ->
+                  {- The grep statement found a package.  -}
+                  pure Other
+                (ExitFailure 1, _, _) ->
+                  {-
+                    grep exited with a exit code 1, meaning no packages were
+                    found, meaning this is an internal module.
+                  -}
+                  pure Local
+                procFailed -> fail (show procFailed)
+
+        contains :: String -> String -> Bool
+        contains (Text.pack -> haystack) (Text.pack -> needle) =
+          fst (Text.breakOn needle haystack) /= haystack
```

Comment:
How about [isInfixOf](https://hoogle.simspace.com/file/home/simspace/.stack/programs/x86_64-linux/ghc-8.6.5/share/doc/ghc-8.6.5/html/libraries/base-4.12.0.0/Data-List.html#v:isInfixOf)?

Reply:
fixed in c21df6e